src/allmydata/test/test_storage_http.py

```diff
@@ -1201,18 +1201,42 @@ def create_upload(self, data=b"abcdef"):
         )
         return storage_index, write_secret, lease_secret
 
-    def test_write_can_be_read(self):
+    def test_write_can_be_read_small_data(self):
+        """
+        Small written data can be read using ``read_share_chunk``.
+        """
+        self.write_can_be_read(b"abcdef")
+
+    def test_write_can_be_read_large_data(self):
+        """
+        Large written data (50MB) can be read using ``read_share_chunk``.
+        """
+        self.write_can_be_read(b"abcdefghij" * 5 * 1024 * 1024)
```

Comment:
These two could be one Hypothesis test.  The "given" could be:

```
@given(sampled_from([b"abcdef", b"abcdefghij" * 5 * 1024 * 1024])
```

for exactly the same behavior as these two tests or something a little more adventurous like:

```
@given(
    integers(min_value=1, max_value=10 * 1024 * 1024).map(
        lambda expansion: b"abcde" * expansion
    )
)
```

to explore the length space between 1 byte and 50MB more exhaustively.

At your discretion.

Reply:
... and that hits the hypothesis deadline thing so I'm just not gonna bother.